zjit.c

```diff
@@ -351,5 +351,16 @@ VALUE rb_zjit_assert_compiles(rb_execution_context_t *ec, VALUE self);
 VALUE rb_zjit_stats(rb_execution_context_t *ec, VALUE self);
 VALUE rb_zjit_stats_enabled_p(rb_execution_context_t *ec, VALUE self);
 
+enum {
+    RB_SPECIAL_CONST_SHAPE_ID = SPECIAL_CONST_SHAPE_ID,
+    RB_INVALID_SHAPE_ID = INVALID_SHAPE_ID,
+};
+
+bool
+rb_zjit_singleton_class_p(VALUE klass)
+{
+    return RCLASS_SINGLETON_P(klass);
+}
```

Comment:
In the comment on L349, I was proposing to not put things like this after L350-352 (similarly to YJIT) because leaving them last makes it clear that these functions declarations are used only by `zjit.rbinc` in the C code.

If you don't like the idea, it's ok, but please update L349 in that case.

Reply:
Sorry, totally missed that! That makes a lot of sense